pil-analyzer/tests/condenser.rs

```diff
@@ -243,3 +243,124 @@ fn new_fixed_column_as_closure() {
     "#;
     analyze_string::<GoldilocksField>(input);
 }
+
+#[test]
+fn set_hint() {
+    let input = r#"
+    namespace std::prover;
+        let set_hint = 8;
+        let eval = 8;
+        enum Query { Hint(fe), None, }
+    namespace N(16);
+        let x;
+        let y;
+        std::prover::set_hint(y, |i| std::prover::Query::Hint(std::prover::eval(x)));
+        {
+            let z;
+            std::prover::set_hint(z, query |_| std::prover::Query::Hint(1));
```

Comment:
can you `set_hint` a column from an outer scope? what about from a function? should those be tested?

Reply:
function or block should no make a difference, but I can add some more tests where the column is "somewhere else"